src/components/App.js

```diff
@@ -0,0 +1,67 @@
+import './App.css'
+import Card from './Card'
+import { useEffect, useState } from 'react'
+import Pagination from './Pagination'
+import getCharacters from '../services/getCharacters'
+
+function App() {
+  const [characters, setCharacters] = useState([])
+  const [cachedCharacters, setCachedCharacters] = useState({})
+  const [totalPages, setTotalPages] = useState(0)
+  const [currentPage, setCurrentPage] = useState(1)
+  const url = 'https://rickandmortyapi.com/api/character?page=' + currentPage
+
+  useEffect(() => {
+    if (!cachedCharacters[currentPage]) {
+      getCharacters(url).then(data => {
+        setCharacters(data.results)
+        setCachedCharacters({
+          ...cachedCharacters,
+          [currentPage]: data.results,
+        })
+        setTotalPages(data.info.pages)
+      })
+    }
+  }, [url])
+
+  function handleClickBack() {
+    setCurrentPage(currentPage - 1)
+    if (cachedCharacters[currentPage - 1]) {
```

Comment:
Do we need to test this? How could there be no cached page? If it was possible to have no cached page here, you should also implement an `else`. TLDR: remove that `if` statement.

Reply:
With clicking on a button, the `currentPage` is updated and because of that the `url`. I added the `url` as a dependency for the `useEffect` hook. So if we don't have any cached pages, then the `useEffect` get's the data with the new `url`. Sounds a bit complicated 😄. I will do as you suggested in the other comment and put that logic in the `useEffect`.